### PR TITLE
examples: Add common Makefile

### PR DESCRIPTION
**Pull request by paddatrapper**
Addresses #21

### CODE REVIEW

File: examples/example.mk [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+$(FX2LIBDIR)/lib/fx2.lib: $(FX2LIBDIR)/.git
+	cd $(dir $@) && make -j1
+
+ 
+# We depend on a file inside the directory as git creates an
+# empty dir for us.
+#
+# Note that although we have the variable FX2LIBDIR, the submodule
+# magic will always check it out in fx2lib/
+$(FX2LIBDIR)/.git: $(GITDIR)
```
Comment (mithro) [inline examples/example.mk]: This line is wrong now. It is suppose to depend on the `.gitmodules` file (which defines the submodule).

File: examples/example.mk [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# To build the firmware you will need:
+#   SDCC from http://sdcc.sourceforge.net
+#   fx2lib from http://fx2lib.wiki.sourceforge.net
```
Comment (mithro) [inline examples/example.mk]: This comment about fx2lib is slightly wrong now.

**Review by mithro — COMMENTED** [review]
I would probably use `common.mk` rather than `example.mk`

**Comment by paddatrapper** [comment]
Fixed

**Comment by paddatrapper** [comment]
@mithro: ping?

**Pull request merged**